src/accounts_refactor.rs

```diff
@@ -0,0 +1,339 @@
+        &self.account
+    }
+}
+
+impl<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> DerefMut
+    for AccountEditor<'_, L, Key>
+{
+    fn deref_mut(&mut self) -> &mut Account<L, Key> {
+        &mut self.account
+    }
+}
+
+/// Accounts stored in an accounts store.
+pub struct Accounts<L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> {
+    /// A key-value store for accounts.
+    store: AccountsStore<L, Key, Key::PubKey>,
+}
+
+impl<L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> Accounts<L, Key> {
+    #![allow(dead_code)]
+
+    /// Load an accounts store.
+    pub fn new(store: AccountsStore<L, Key, Key::PubKey>) -> Result<Self, KeystoreError<L>> {
+        Ok(Self { store })
+    }
+
+    /// Iterate through the accounts.
+    pub fn iter(&self) -> impl Iterator<Item = Account<L, Key>> + '_ {
+        self.store.iter().cloned()
+    }
+
+    /// Get the account by the public key from the store.
+    pub fn get(&self, pub_key: &Key::PubKey) -> Result<Account<L, Key>, KeystoreError<L>> {
+        Ok(self.store.load(pub_key)?)
+    }
+
+    /// Get a mutable account editor by the public key from the store.
+    pub fn get_mut(
+        &mut self,
+        pub_key: &Key::PubKey,
+    ) -> Result<AccountEditor<L, Key>, KeystoreError<L>> {
+        let account = self.get(pub_key)?;
+        Ok(AccountEditor::new(&mut self.store, account))
+    }
+
+    /// Commit the store version.
+    pub fn commit(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.commit_version()?)
+    }
+
+    /// Revert the store version.
+    pub fn revert(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.revert_version()?)
+    }
+
+    /// Create an account.
+    ///
+    /// Returns the editor for the created account.
+    pub fn create(
+        &mut self,
+        key: Key,
+        description: String,
+        assets: Vec<AssetCode>,
+        records: Vec<RecordInfo>,
+    ) -> Result<AccountEditor<L, Key>, KeystoreError<L>> {
+        let mut balances = HashMap::new();
+        for rec in &records {
+            *balances
+                .entry(rec.ro.asset_def.code)
+                .or_insert_with(U256::zero) += rec.amount().into();
+        }
+        let time = Local::now();
+        let account = Account {
+            key,
+            description,
+            used: false,
+            scan: None,
+            assets,
+            records,
+            balances,
+            created_time: time,
+            modified_time: time,
+        };
+        let mut editor = AccountEditor::new(&mut self.store, account);
+        editor.save()?;
+        Ok(editor)
+    }
+
+    /// Update the ledger scan of an account.
+    pub(crate) async fn update_scan(
+        &mut self,
+        pub_key: &Key::PubKey,
+        event: LedgerEvent<L>,
+        source: EventSource,
+        records_commitment: MerkleCommitment,
+    ) -> Result<AccountEditor<'_, L, Key>, KeystoreError<L>> {
```

Comment:
Same comment here about returning the results of the scan.

Reply:
Unclear that this function is necessary since `model.get_mut(primary_key).do_something().save()` is the idiomatic way to update a resource anyways, and this function just does exactly that